src/thunderbots/software/ai/hl/stp/play/enemy_freekick_play.cpp

```diff
@@ -0,0 +1,158 @@
+
+        // Check if the enemy is passing-capable
+        bool enemy_team_can_pass =
+            Util::DynamicParameters::EnemyCapability::enemy_team_can_pass.value();
+
+        // Update goalie tactic
+        goalie_tactic->updateParams(world.ball(), world.field(), world.friendlyTeam(),
+                                    world.enemyTeam());
+
+        // Update free kicke shadowers
+        shadow_freekicker_1->updateParams(world.enemyTeam(), world.ball());
+        shadow_freekicker_2->updateParams(world.enemyTeam(), world.ball());
+
+        // Update crease defenders
+        crease_defender_tactic->updateParams(world.ball(), world.field(),
+                                             world.friendlyTeam(), world.enemyTeam());
+
+        // Assign ShadowEnemy tactics until we have every enemy covered. If there are not
+        // enough threats to shadow, move our robots to block the friendly net
+        if (enemy_threats.size() == 0)
+        {
+            move_tactic_main->updateParams(
+                world.field().friendlyGoal() + Point(0, 2 * ROBOT_MAX_RADIUS_METERS),
+                (world.ball().position() - world.field().friendlyGoal()).orientation(),
+                0);
+            move_tactic_main->updateParams(
+                world.field().friendlyGoal() + Point(0, -2 * ROBOT_MAX_RADIUS_METERS),
+                (world.ball().position() - world.field().friendlyGoal()).orientation(),
+                0);
+
+            tactics_to_run.emplace_back(move_tactic_main);
+            tactics_to_run.emplace_back(move_tactic_secondary);
+        }
+        if (enemy_threats.size() == 1)
+        {
+            shadow_tactic_main->updateParams(
+                enemy_threats.at(1), world.field(), world.friendlyTeam(),
+                world.enemyTeam(), ROBOT_MAX_RADIUS_METERS * 3, enemy_team_can_pass);
+            move_tactic_main->updateParams(
+                world.field().friendlyGoal() + Point(0, 2 * ROBOT_MAX_RADIUS_METERS),
+                (world.ball().position() - world.field().friendlyGoal()).orientation(),
+                0);
+
+            tactics_to_run.emplace_back(shadow_tactic_main);
+            tactics_to_run.emplace_back(move_tactic_main);
+        }
+        if (enemy_threats.size() >= 2)
+        {
+            shadow_tactic_main->updateParams(
+                enemy_threats.at(1), world.field(), world.friendlyTeam(),
+                world.enemyTeam(), ROBOT_MAX_RADIUS_METERS * 3, enemy_team_can_pass);
+            shadow_tactic_secondary->updateParams(
+                enemy_threats.at(2), world.field(), world.friendlyTeam(),
+                world.enemyTeam(), ROBOT_MAX_RADIUS_METERS * 3, enemy_team_can_pass);
+
+            tactics_to_run.emplace_back(shadow_tactic_main);
+            tactics_to_run.emplace_back(shadow_tactic_secondary);
+        }
+
+
+
+        tactics_to_run.emplace_back(crease_defender_tactic);
```

Comment:
Are these lowest priority? Right now they are, but crease defender is probably higher priority?